nncf/quantization/algorithms/weight_compression/awq.py

```diff
@@ -150,19 +145,30 @@ def apply(
             if not self._backend_entity.is_node_with_weights(nncf_node, graph):
                 continue
 
+            target_node_names = []
             for weight_op_friendly_name, _ in self._backend_entity.get_weight_names_and_port_ids(nncf_node, graph):
                 target_node_names.append(weight_op_friendly_name)
 
-            nncf_node = graph.get_node_by_key(match[0])
-            for weight_op_friendly_name, _ in self._backend_entity.get_weight_names_and_port_ids(nncf_node, graph):
-                merge_node_names.append(weight_op_friendly_name)
-
-            assert len(target_node_names) == len(merge_node_names)
             weight_params = self._all_weight_params[name_mapping[target_node_names[-1]]]
+
             if weight_params.compression_config.num_bits != 4:
                 continue
             target_node = self._nodes_to_compress[name_mapping[target_node_names[-1]]]
-            merge_node = self._nodes_to_compress[name_mapping[merge_node_names[-1]]]
+
+            # avoid matching different patterns for the same node
+            if target_node.node_name in awq_data:
+                continue
+
+            nncf_node = graph.get_node_by_key(match[0])
+
+            if self._backend_entity.is_node_with_weights(nncf_node, graph):  # pattern MatMul->Multiply->MatMul
+                merge_node_names = []
+                for weight_op_friendly_name, _ in self._backend_entity.get_weight_names_and_port_ids(nncf_node, graph):
+                    merge_node_names.append(weight_op_friendly_name)
+                assert len(target_node_names) == len(merge_node_names)
```

Comment:
per offline discussion, should be deleted, since the matmul can't have more than one weight on input

Reply:
Done